### ISSUE TITLE
Framework-Dependent Deployment not loading assemblies on linux-arm

### ISSUE THREAD

**Issue by pierslawson** [dotnet/coreclr#17438]
I'm trying to use .Net Core 2.1 Preview 1 on a Raspberry Pi so publishing with -r linux-arm. I've followed the [omnisharp guidelines for remote debugging](https://github.com/OmniSharp/omnisharp-vscode/wiki/Remote-Debugging-On-Linux-Arm) and can successfully step into code using the 
Framework-Dependent Deployment approach if I have a simple console application. Brilliant!

However, if I add a reference to the project to a nuget package and try to debug I get missing assembly messages. For example, adding:
```
<PackageReference Include="System.Text.Encoding.CodePages" Version="4.4.0" />
```
to the csproj and building results in the following message when I try to debug or run directly (```sudo dotnet MyConsoleApp.dll```):
```
Error:
An assembly specified in the application dependencies manifest (MyConsoleApp.deps.json) was not found:
package: 'System.Text.Encoding.CodePages', version: '4.4.0'
path: 'lib/netstandard2.0/System.Text.Encoding.CodePages.dll'
The target process exited without raising a CoreCLR started event. Ensure that the target process is configured to use .NET Core. This may be expected if the target process did not run on .NET Core.
```
If I publish the project, the published version does execute... however I can't currently debug the published version.

I running on an RPi3:
```
PRETTY_NAME="Raspbian GNU/Linux 9 (stretch)"
NAME="Raspbian GNU/Linux"
VERSION_ID="9"
VERSION="9 (stretch)"
ID=raspbian
ID_LIKE=debian
HOME_URL="http://www.raspbian.org/"
SUPPORT_URL="http://www.raspbian.org/RaspbianForums"
BUG_REPORT_URL="http://www.raspbian.org/RaspbianBugs"
```
Running dotnet 2.1 Preview 1:
```
Microsoft .NET Core Shared Framework Host
  Version  : 2.1.0-preview1-26216-03
  Build    : f2c3216183d20416568a4bbf5bb7d153e826f153
```

**Comment by RussKeldorph** [comment]
@dotnet/dotnet-diag If this appears to work as intended from a debugging point of view, perhaps it should go to dotnet/cli.

/cc @livarcocc

**Comment by pierslawson** [comment]
Yes, by directly I meant from a command prompt on the Raspberry Pi a ran the cli from a local directory containing the code (I actually have the code residing on the RPi and shared using Samba with the Windows machine which is doing the compiling. 

I have recreated this issue in the dotnet/cli#9039.

The inability to debug the published version I have already raised within OmniSharp/omnisharp-vscode#2150 and I think @gregg-miskelly has seen it.
